client/post-editor/index.js

```diff
@@ -14,6 +14,7 @@ module.exports = function() {
 	page( '/post', controller.pressThis, sitesController.siteSelection, sitesController.sites );
 	page( '/post/new', () => page.redirect( '/post' ) ); // redirect from beep-beep-boop
 	page( '/post/:site?/:post?', sitesController.siteSelection, sitesController.fetchJetpackSettings, controller.post );
+	page( '/post/revision/:site?/:post?', sitesController.siteSelection, sitesController.fetchJetpackSettings, controller.revision );
```

Comment:
> will do :)

or maybe not ^^. It seems this breaks the expected "position" of the site ID in `getSiteFragment`: https://github.com/Automattic/wp-calypso/blob/master/client/lib/route/path.js#L23

So we should either stick to `/post/revisions/:site/:post/` or maybe `/revision/:site/:post`. Last solution could be to attack the root cause and get `getSiteFragment` to use page.js's params first and then default to exploding the path. It seems most routes are using `:site`, I've found only 3 occurences of `:siteId`, there could be other weird usages out there so this feels like a risky move tbh.